File: core-mvp-sample/src/main/java/ru/surfstudio/android/core/mvp/sample/ui/base/configurator/ActivityConfigurator.java
package ru.surfstudio.android.core.mvp.sample.ui.base.configurator;

import ru.surfstudio.android.core.mvp.sample.app.App;
import ru.surfstudio.android.core.mvp.sample.app.dagger.AppComponent;
import ru.surfstudio.android.core.mvp.sample.ui.base.dagger.activity.ActivityModule;
import ru.surfstudio.android.core.mvp.sample.ui.base.dagger.activity.DaggerActivityComponent;
import ru.surfstudio.android.core.ui.configurator.BaseActivityConfigurator;
import ru.surfstudio.android.core.mvp.sample.ui.base.dagger.activity.ActivityComponent;

/**
 * Базовый конфигуратор для активити
 */

public class ActivityConfigurator extends BaseActivityConfigurator<ActivityComponent, AppComponent> {

    @Override
    protected ActivityComponent createActivityComponent(AppComponent parentComponent) {
        return DaggerActivityComponent.builder()
                .appComponent(parentComponent)
                .activityModule(new ActivityModule(getPersistentScope()))
                .build();
    }

    @Override
    protected AppComponent getParentComponent() {
        return ((App) getTargetActivity().getApplicationContext()).getAppComponent();
    }
}

File: template-multimodule/app-injector/src/main/java/ru/surfstudio/standard/app_injector/ui/configurator/ActivityConfigurator.kt
package ru.surfstudio.standard.app_injector.ui.configurator

import android.content.Context
import ru.surfstudio.android.core.ui.configurator.BaseActivityConfigurator
import ru.surfstudio.standard.app_injector.ui.ActivityComponent
import ru.surfstudio.standard.app_injector.ui.DaggerActivityComponent

class ActivityConfigurator(val context: Context) :
        BaseActivityConfigurator<ActivityComponent, ru.surfstudio.standard.app_injector.AppComponent>() {

    override fun createActivityComponent(parentComponent: ru.surfstudio.standard.app_injector.AppComponent?): ActivityComponent =
            DaggerActivityComponent.builder()
                    .appComponent(parentComponent)
                    .build()

    override fun getParentComponent(): ru.surfstudio.standard.app_injector.AppComponent =
            (context.applicationContext as ru.surfstudio.standard.app_injector.App).appComponent
}